dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/dataapproval/DataApprovalServiceTest.java

```diff
@@ -1100,18 +1057,14 @@ public void testGetDataApprovalStateAboveUserOrgUnitLevel()
     @Test
     public void testGetDataApprovalStateOrgUnitAssignments()
     {
-        Set<OrganisationUnit> units = newHashSet( organisationUnitA );
-
-        CurrentUserService currentUserService = new MockCurrentUserService( units, null, DataApproval.AUTH_APPROVE,
```

Comment:
Note that the user creation and switch was moved below updaing the dataset for this case. The reason is that the `new MockCurrentUserService` as used here and elsewhere did create a superuser. As such that user also could update the dataset. I assume this was unintentional so the users created via `createUserAndInjectSecurityContext` where not made superusers. To still be able to update the dataset as part of the test-setup here the user is first switched after the dataset is updated.